examples/hello/server.c

```diff
@@ -0,0 +1,171 @@
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_pt_index_t pti;
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_le_t leh;
+	long num_messages;
+	ptl_le_t le;
+	char *p;
+	char msg[PTL_EV_STR_SIZE];
+	char data[50];
+	bool wait_for_ever = false;
+
+	if (argc < 2) {
+		wait_for_ever = true;
+	} else {
+		num_messages = strtol(argv[1], &p, 10);
+		if (*p != '\0') {
+			fprintf(stderr,
+				"An invalid character was found before the end of the string");
+			return 1;
+		}
+	}
+
+	printf(" \n--- server interface --- \n \n");
+
+	ret = PtlInit();
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlNIInit(PTL_IFACE_DEFAULT, PTL_NI_NO_MATCHING | PTL_NI_PHYSICAL, SERVER_PID, NULL,
+			NULL, &nih);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto fini;
+	}
+
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlPTAlloc(nih, 0, eqh, 0, &pti);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto eq_free;
+	}
+
+	/* display server pid and server nid so the client knows the server id */
+	ret = PtlGetId(nih, &id);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto le_unlink;
+	}
+	printf("Server Pid : %d\n", id.phys.pid);
+	printf("Server Nid : %d\n\n", id.phys.nid);
+
+	/* describe the memory used to receive the message */
+	le = (ptl_le_t){ .start = data,
+			 .length = sizeof(data),
+			 .ct_handle = PTL_CT_NONE,
+			 .uid = PTL_UID_ANY,
+			 .options = PTL_LE_OP_PUT };
+
+	/* expose receive buffer to the network */
+	ret = PtlLEAppend(nih, pti, &le, PTL_PRIORITY_LIST, NULL, &leh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto le_unlink;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s", msg);
+	printf("Server is ready\n");
+
+	while (num_messages > 0 || wait_for_ever) {
+		/* wait until a message arrives */
+		printf("\n");
+		ret = PtlEQWait(eqh, &ev);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			res = 1;
+			goto le_unlink;
+		}
+		data[sizeof(data) - 1] = '\0';
+		PtlEvToStr(0, &ev, msg);
+		printf("Event : %s", msg);
+		if (ev.type == PTL_EVENT_PUT) {
+			if (ev.rlength + 1 >= sizeof(data)) {
+				fprintf(stderr,
+					"The received message is too long to be entirely readed\n");
+			} else {
+				((char *)ev.start)[ev.rlength + 1] = 0;

```

Comment:
I don't think this is needed.
The \0 final string character has been transmitted as part of the message.